### PR TITLE
HUGCHAT Tool

### PR DESCRIPTION
**Pull request by Aminolroaya**
I added unit test for hugchat tool, integration test can also be added. Hugchat will not work currently without cookies.

### CODE REVIEW

**Comment by 20001LastOrder** [comment]
Thanks for the PR. could you provide a bit more information (maybe a link) about how I can find the environment variables for hugging face and test it locally?

**Comment by 20001LastOrder** [comment]
Nevermind, I realized it's just the password and email...

File: src/tests/unit_tests/tools/test_hug_chat_tool.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from sherpa_ai.tools import HugChatTool
```
Comment (20001LastOrder) [inline src/tests/unit_tests/tools/test_hug_chat_tool.py:1]: Please add a new line at the end and add `@external_api` before the test

File: src/sherpa_ai/tools.py [hunk 2]
```diff
@@ -209,3 +218,31 @@ def _run(self, query: str) -> str:
...
+    )
+    def _run(self, query: str) -> str:
+
+        # Log in to huggingface and grant authorization to huggingchat
+        sign = Login(cfg.HUGCHAT_EMAIL, cfg.HUGCHAT_PASS)
+        # Save cookies to the local directory
+        cookies = sign.login()
+
+        # Save cookies to the local directory
+        cookie_path_dir = "./cookies_snapshot"
```
Comment (20001LastOrder) [inline src/sherpa_ai/tools.py:236]: I think even if we cannot avoid the cookies, we don't need to save them locally. We can remove line 236 and 237

**Review by 20001LastOrder — CHANGES_REQUESTED** [review]
Thanks for the PR. Just some minor comments, please see them below.

**Comment by oshoma** [comment]
Hi @Aminolroaya @20001LastOrder  I made some changes as follows:

- enable offline testing with omcks
- enable online testing 
- reformat 
- get Poetry in synch with `main` 

Could you both please try running the following tests to be sure everything works and report back here? If you have Hugging Chat credentials in your .env file then you should get 11 passing tests. Without credentials you should get 10 passing tests and 1 skipped test.

```
poetry install
pytest tests/unit_tests/tools/test_hug_chat_tool.py   -v
```

And if that works, try `pytest tests` too. Here's my output: `1 failed, 56 passed, 2 skipped, 13 warnings`. 
The failure is expected, it will be resolved when we fix #252.

@20001LastOrder If tests run clean and your code review is OK then I think with these changes we are ready to merge into `main`. We should do a squash merge.

**Comment by Aminolroaya** [comment]
@oshoma apologies for the late reply. Is there anything that needs to be taken care of before merging? cc @amirfz

**Comment by amirfz** [comment]
@Aminolroaya I think there are some unaddressed comments from Osh on this PR. have you seen those?

**Comment by Aminolroaya** [comment]
> @Aminolroaya I think there are some unaddressed comments from Osh on this PR. have you seen those?

Yes. Just want to check if using the Hugchat tool is still considered an enhancement for the project. If yes, I will look into it this weekend. tnx.

**Comment by 20001LastOrder** [comment]
@Aminolroaya I think this is definitely still valuable. Especially now we are planning on fully configurable agents, adding a new tool to an agent will be much easier. Thus, an additional tool would be nice for different use cases.

**Comment by 20001LastOrder** [comment]
Closed as not planned

**Pull request closed without merging**